### PR TITLE
Availability backend

### PR DESCRIPTION
**Pull request by magsyg**
Added availability class for recruitment

Went with that they could fill out the times they werent available, instead of times they were available since I think that is less entries for a student, and can be better to operate on for a view of finding available applicant times

Only backend for posting and fetching
Needs probably be improved, but will be done when its logic is more or less refined

### CODE REVIEW

File: backend/samfundet/models/recruitment.py [hunk 1]
```diff
@@ -201,3 +201,21 @@ def save(self, *args: tuple, **kwargs: dict) -> None:
...
+
+    user = models.ForeignKey(
+        User,
+        on_delete=models.CASCADE,
+        null=False,
+        blank=False,
+        help_text='Occupied timeslots for user',
+        related_name='occupied_timeslots',
+    )
+    # Mostly only used for deletion, and anonomization
```
Comment (emilte) [inline backend/samfundet/models/recruitment.py:216]:
```suggestion
    # Mostly only used for deletion, and anonymization.
```

File: backend/samfundet/models/recruitment.py [hunk 2]
```diff
@@ -201,3 +201,21 @@ def save(self, *args: tuple, **kwargs: dict) -> None:
                 self.interview = Interview.objects.create()
 
         super().save(*args, **kwargs)
+
+
+class OccupiedTimeslots(FullCleanSaveMixin):
```
Comment (emilte) [inline backend/samfundet/models/recruitment.py:206]:
```suggestion
class OccupiedTimeslot(FullCleanSaveMixin):
```

**Review by Snorre98 — APPROVED** [review]
LGTM :)

File: backend/samfundet/models/recruitment.py [hunk 3]
```diff
@@ -201,3 +201,21 @@ def save(self, *args: tuple, **kwargs: dict) -> None:
...
+        blank=False,
+        help_text='Occupied timeslots for user',
+        related_name='occupied_timeslots',
+    )
+    # Mostly only used for deletion, and anonomization
+    recruitment = models.ForeignKey(Recruitment, on_delete=models.CASCADE, help_text='Occupied timeslots for the users for this recruitment')
+
+    # Start and end time of availability
+    start_time = models.DateTimeField(help_text='The time of the interview', null=False, blank=False)
+    end_time = models.DateTimeField(help_text='The time of the interview', null=False, blank=False)
```
Comment (emilte) [inline backend/samfundet/models/recruitment.py:221]: Litt enklere å vite hva feltet inneholder.
```suggestion
    start_dt = models.DateTimeField(help_text='The time of the interview', null=False, blank=False)
    end_dt = models.DateTimeField(help_text='The time of the interview', null=False, blank=False)
```

File: backend/samfundet/admin.py [hunk 4]
```diff
@@ -62,6 +63,8 @@
 
 # Unregister User and Group to set new Admins.
 admin.site.unregister(Group)
+# Just for testing TODO remove when done
+admin.site.register(OccupiedTimeslots)
```
Comment (emilte) [inline backend/samfundet/admin.py:67]: Den kan sikkert bare være der, eller vil du ikke?

Comment (magsyg) [reply backend/samfundet/admin.py:67]: Mer for at det er privat data, vi har egt ikke noe vi kommer til å gjøre med det, mer et subobjekt en annet

File: backend/samfundet/models/recruitment.py (hunk 3 above)
Comment (magsyg) [reply backend/samfundet/models/recruitment.py]: Done

File: backend/samfundet/models/recruitment.py (hunk 2 above)
Comment (magsyg) [reply backend/samfundet/models/recruitment.py]: Done

File: backend/samfundet/models/recruitment.py (hunk 1 above)
Comment (magsyg) [reply backend/samfundet/models/recruitment.py]: Done

File: backend/samfundet/urls.py [hunk 5]
```diff
@@ -62,4 +62,5 @@
     path('recruitment-positions/', views.RecruitmentPositionsPerRecruitmentView.as_view(), name='recruitment_positions'),
     path('active-recruitment-positions/', views.ActiveRecruitmentPositionsView.as_view(), name='active_recruitment_positions'),
     path('applicants-without-interviews/', views.ApplicantsWithoutInterviewsView.as_view(), name='applicants_without_interviews/'),
+    path('Occupiedtimeslot/', views.OccupiedtimeslotView.as_view(), name='occupied_timeslots')
```
Comment (emilte) [inline backend/samfundet/urls.py:65]:
```suggestion
    path('occupiedtimeslot/', views.OccupiedtimeslotView.as_view(), name='occupied_timeslots')
```

**Review by emilte — APPROVED** [review]

**Pull request merged**